src/subdomains/core/payment-link/controllers/payment-link.gateway.ts

```diff
@@ -0,0 +1,47 @@
+import { BadRequestException } from '@nestjs/common';
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { OnGatewayConnection, WebSocketGateway } from '@nestjs/websockets';
+import { IncomingMessage } from 'http';
+import { Lock } from 'src/shared/utils/lock';
+import { Util } from 'src/shared/utils/util';
+
+type ClientMap = Map<string, Map<string, WebSocket>>;
+
+@WebSocketGateway({ path: '/v1/paymentLink' })
+export class PaymentLinkGateway implements OnGatewayConnection {
+  private readonly clients: ClientMap = new Map();
+
+  @Cron(CronExpression.EVERY_30_SECONDS)
+  @Lock()
+  sendMessages() {
+    for (const devices of this.clients.values()) {
+      for (const client of devices.values()) {
+        client.send('12-10000');
+      }
+    }
+  }
+
+  handleConnection(client: WebSocket, message: IncomingMessage) {
+    const device = new URLSearchParams(message.url?.split('?')[1]).get('device');
+    if (!device) throw new BadRequestException('device should not be empty');
```

Comment:
`Device`

Reply:
It's actually adapted to the automated validation error messages
<img width="345" alt="image" src="https://github.com/user-attachments/assets/7bf0ed8d-9cc6-4e36-977e-38ae3cc4cd8b">

Reply:
Haha, yes, maybe 🤔